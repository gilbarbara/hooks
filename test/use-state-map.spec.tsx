import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';

import useStateMap from '../src/use-state-map';

describe('useStateMap', () => {
  const Component = () => {
    const [options, setOptions] = useStateMap({ data: [], status: 'idle' });

    const handleClick = React.useCallback(() => {
      setOptions({ status: 'running' });
    }, [setOptions]);

    return (
      <div>
        <h1>{JSON.stringify(options, null, 2)}</h1>
        <button type="button" onClick={handleClick}>
          Update
        </button>
      </div>
    );
  };

  const { container } = render(<Component />);

  it('should re-render the id has changed', () => {
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(container.querySelector('button')!);

    expect(container.firstChild).toMatchSnapshot();
  });
});
