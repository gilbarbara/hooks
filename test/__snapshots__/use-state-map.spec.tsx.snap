// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useStateMap should re-render the id has changed 1`] = `
<div>
  <h1>
    {
  "data": [],
  "status": "idle"
}
  </h1>
  <button
    type="button"
  >
    Update
  </button>
</div>
`;

exports[`useStateMap should re-render the id has changed 2`] = `
<div>
  <h1>
    {
  "data": [],
  "status": "running"
}
  </h1>
  <button
    type="button"
  >
    Update
  </button>
</div>
`;
